rgbd/src/get_3d_point_from_image_roi_node.cpp

```diff
@@ -1,151 +1,154 @@
 #include <boost/circular_buffer.hpp>
 
 // ----------------------------------------------------------------------------------------------------
 
-boost::circular_buffer<std::shared_ptr<rgbd::Image>> g_last_images_;
+boost::circular_buffer<std::shared_ptr<rgbd::Image> > g_last_images_;
 bool srvGet3dPointFromROI(rgbd_msgs::Project2DTo3D::Request& req, rgbd_msgs::Project2DTo3D::Response& res)
 {
-  std::shared_ptr<rgbd::Image> last_image;
+    std::shared_ptr<rgbd::Image> last_image;
 
-  if (!g_last_images_.empty())
-  {
-    if (req.stamp == ros::Time(0))
-    {
-      last_image = g_last_images_.back();
-    }
-    else
+    if (!g_last_images_.empty())
     {
-      for (auto it = g_last_images_.rbegin(); it != g_last_images_.rend(); ++it)
-      {
-        if ( (*it)->getTimestamp() <= req.stamp.toSec() )
+        if (req.stamp == ros::Time(0))
+            last_image = g_last_images_.back();
+        else
         {
-          last_image = *it;
+            for (auto it = g_last_images_.rbegin(); it != g_last_images_.rend(); ++it)
+            {
+                if ( (*it)->getTimestamp() <= req.stamp.toSec())
+                    last_image = *it;
+            }
         }
-      }
     }
-  }
 
-  if (!last_image) {
-    ROS_ERROR("I could not find an images in my image buffer for timestamp %.2f", req.stamp.toSec());
-    return false;
-  }
+    if (!last_image)
+    {
+        ROS_ERROR("I could not find an images in my image buffer for timestamp %.2f", req.stamp.toSec());
+        return false;
+    }
 
-  for (const sensor_msgs::RegionOfInterest& roi : req.rois)
-  {
-    const cv::Mat& depth = last_image->getDepthImage();
+    for (const sensor_msgs::RegionOfInterest& roi : req.rois)
+    {
+        const cv::Mat& depth = last_image->getDepthImage();
 
-    cv::Rect roi_rgb(roi.x_offset, roi.y_offset, roi.width, roi.height);
-    float rgb_depth_width_ratio = (float)(depth.cols) / last_image->getRGBImage().cols;
-    cv::Rect roi_depth(rgb_depth_width_ratio * roi_rgb.tl(), rgb_depth_width_ratio * roi_rgb.br());
-    cv::Point roi_depth_center = 0.5 * (roi_depth.tl() + roi_depth.br());
+        cv::Rect roi_rgb(static_cast<int>(roi.x_offset), static_cast<int>(roi.y_offset),
+                         static_cast<int>(roi.width), static_cast<int>(roi.height));
+        float rgb_depth_width_ratio = static_cast<float>(depth.cols) / static_cast<float>(last_image->getRGBImage().cols);
+        cv::Rect roi_depth(rgb_depth_width_ratio * roi_rgb.tl(), rgb_depth_width_ratio * roi_rgb.br());
+        cv::Point roi_depth_center = 0.5 * (roi_depth.tl() + roi_depth.br());
 
-    cv::Rect roi_depth_capped(cv::Point(std::max(0, roi_depth.x),
-                                        std::max(0, roi_depth.y)),
-                              cv::Point(std::min(depth.cols - 1, roi_depth.br().x),
-                                        std::min(depth.rows - 1, roi_depth.br().y)));
+        cv::Rect roi_depth_capped(cv::Point(std::max(0, roi_depth.x),
```

Comment:
I don't think this improves readability

Reply:
That wasn't my intention, restored the indent.